tools/pipelines/build-docs.yml

```diff
@@ -282,18 +285,63 @@ stages:
               pnpm i --frozen-lockfile
 
         - task: Npm@1
-          displayName: npm run build
+          displayName: Build
+          inputs:
+            command: 'custom'
+            workingDir: $(Build.SourcesDirectory)/docs
+            customCommand: 'run build'
+
+        - task: Npm@1
+          displayName: Validate links
+          inputs:
+            command: 'custom'
+            workingDir: $(Build.SourcesDirectory)/docs
+            customCommand: 'run check-links'
+
+# TODO: this redundantly builds the website.
```

Comment:
I'd even argue running the tests could be part of the first build stage/job, so we only publish a build artifact is the tests pass.

Reply:
I guess we're less strict about the link checking. Maybe that's a better short -term solution. Still, arguably we should be able to have our post-build jobs run in parallel, rather than sticking everything in a single job.

Reply:
Updated. https://github.com/microsoft/FluidFramework/pull/22990/commits/4f8b4fb675672364f0bf4da3cb6e4777b33ea727